src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/ModelMetadataUtilities.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Web.CodeGeneration;
+using Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore;
+
+namespace Microsoft.VisualStudio.Web.CodeGenerators.Mvc
+{
+    internal static class ModelMetadataUtilities
+    {
+        internal static async Task<ModelTypeAndContextModel> ValidateModelAndGetCodeModelMetadata(
+            CommonCommandLineModel commandLineModel, 
+            IEntityFrameworkService entityFrameworkService, 
+            IModelTypesLocator modelTypesLocator)
+        {
+            ModelType model = ValidationUtil.ValidateType(commandLineModel.ModelClass, "model", modelTypesLocator);
+
+            Contract.Assert(model != null, MessageStrings.ValidationSuccessfull_modelUnset);
+            var result = await entityFrameworkService.GetModelMetadata(model);
+
+            return new ModelTypeAndContextModel()
+            {
+                ModelType = model,
+                ContextProcessingResult = result
+            };
+        }
+
+        internal static async Task<ModelTypeAndContextModel> ValidateModelAndGetMetadata(
```

Comment:
>ValidateModelAndGetMetadata [](start = 61, length = 27)

These two method names are similar and not clear what is the difference since result is same type. I would rename them to make sure it is clear that second method returns EF data too